homeassistant/components/sensor/ebusd.py

```diff
@@ -0,0 +1,254 @@
+    'Zone1ActualRoomTemperatureDesired':
+        ['z1ActualRoomTempDesired', '°C', 'mdi:thermometer', 0],
+    'Zone1TimerMonday':
+        ['z1Timer.Monday', None, 'mdi:timer', 1],
+    'Zone1TimerTuesday':
+        ['z1Timer.Tuesday', None, 'mdi:timer', 1],
+    'Zone1TimerWednesday':
+        ['z1Timer.Wednesday', None, 'mdi:timer', 1],
+    'Zone1TimerThursday':
+        ['z1Timer.Thursday', None, 'mdi:timer', 1],
+    'Zone1TimerFriday':
+        ['z1Timer.Friday', None, 'mdi:timer', 1],
+    'Zone1TimerSaturday':
+        ['z1Timer.Saturday', None, 'mdi:timer', 1],
+    'Zone1TimerSunday':
+        ['z1Timer.Sunday', None, 'mdi:timer', 1],
+    'Zone1OperativeMode':
+        ['z1OpMode', None, 'mdi:math-compass', 3],
+    'ContinuosHeating':
+        ['ContinuosHeating', '°C', 'mdi:weather-snowy', 0],
+    'PowerEnergyConsumptionLastMonth':
+        ['PrEnergySumHcLastMonth', 'kWh', 'mdi:flash', 0],
+    'PowerEnergyConsumptionThisMonth':
+        ['PrEnergySumHcThisMonth', 'kWh', 'mdi:flash', 0]
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_CIRCUIT): cv.string,
+    vol.Required(CONF_HOST): cv.string,
+    vol.Optional(CONF_PORT, default=DEFAULT_PORT): cv.port,
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_MONITORED_CONDITIONS, default=[]):
+        vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Set up the Ebusd..."""
+    name = config.get(CONF_NAME)
+    circuit = config.get(CONF_CIRCUIT)
+    server_address = (config.get(CONF_HOST), config.get(CONF_PORT))
+
+    try:
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        data = EbusdData(server_address, circuit)
+
+        sock.settimeout(5)
+        sock.connect(server_address)
+        sock.close()
+
+        dev = []
+        for variable in config[CONF_MONITORED_CONDITIONS]:
+            dev.append(Ebusd(data, variable, name))
+
+        add_devices(dev)
+        hass.services.register('sensor', 'ebusd_write', data.write)
+    except socket.timeout:
+        raise PlatformNotReady
+    except socket.error:
+        raise PlatformNotReady
+
+
+def timer_format(string):
+    """Datetime formatter."""
+    _r = []
+    _s = string.split(';')
+    for i in range(0, len(_s) // 2):
+        if(_s[i * 2] != '-:-' and _s[i * 2] != _s[(i * 2) + 1]):
+            _r.append(_s[i * 2] + '/' + _s[(i * 2) + 1])
+    return ' - '.join(_r)
+
+
+class EbusdData:
+    """Get the latest data from Ebusd."""
+
+    def __init__(self, address, circuit):
+        """Initialize the data object."""
+        self._circuit = circuit
+        self._address = address
+        self.value = {}
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self, name):
+        """Call the Ebusd API to update the data."""
+        import ebusdpy
+        command = READ_COMMAND.format(self._circuit, name, CACHE_TTL)
```

Comment:
This should be in ebusdpy.

Reply:
Doesn't male sense to have it in ebusdpy library. It should contain only the methods to read/write from socket